bitwarden_license/src/Commercial.Infrastructure.EntityFramework/SecretsManager/Repositories/AccessPolicyRepository.cs

```diff
@@ -157,6 +157,21 @@ public async Task ReplaceAsync(Core.SecretsManager.Entities.BaseAccessPolicy bas
         }
     }
 
+    public async Task<IEnumerable<Core.SecretsManager.Entities.BaseAccessPolicy>?> GetManyByGrantedServiceAccountIdAsync(Guid id)
+    {
+        using var scope = ServiceScopeFactory.CreateScope();
+        var dbContext = GetDatabaseContext(scope);
+
+        var entities = await dbContext.AccessPolicies.Where(ap =>
+                ((UserServiceAccountAccessPolicy)ap).GrantedServiceAccountId == id ||
+                ((GroupServiceAccountAccessPolicy)ap).GrantedServiceAccountId == id)
+            .Include(ap => ((UserServiceAccountAccessPolicy)ap).OrganizationUser.User)
+            .Include(ap => ((GroupServiceAccountAccessPolicy)ap).Group)
+            .ToListAsync();
+
+        return !entities.Any() ? null : entities.Select(MapToCore);
```

Comment:
Do we want to return an empty list to avoid a null check?

Reply:
Resolved in https://github.com/bitwarden/server/pull/2649/commits/3f174b966fe69c6b44442b3798474162000c1a11